### ISSUE TITLE
Detect missing `else` in block with `return`

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust#135857]
```
error: expected identifier, found keyword `return`
   --> /home/gh-estebank/rust/compiler/rustc_lint/src/default_could_be_derived.rs:249:13
    |
248 |         let hir::VariantData::Struct { fields, recovered: Recovered::No } = data {
    |                                                                             ---- while parsing this struct
249 |             return;
    |             ^^^^^^ expected identifier, found keyword
    |
help: escape `return` to use it as an identifier
    |
249 |             r#return;
    |             ++
```

This should detect that there's a missing `else`: `= data else {`.

**Comment by 11happy** [comment]
Hello @estebank I am new to Rust community & this issue looks interesting to me. Can I work on this ?
Thank you

**Comment by estebank** [comment]
@11happy you can! You should look at the parser for this. You can look at the docs https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html to set up your environment and after that you can use `rustc +your-toolchain file.rs -Ztrack-diagnostics` to see where the error is emitted.

**Comment by 11happy** [comment]
Sure! Thank you for the headstart.

@rustbot claim

**Comment by bb1yd** [comment]
@11happy Are you still working on this issue?

**Comment by bb1yd** [comment]
I would like to work on this

@rustbot claim
